core/lib/resources/presence/presence_manager.js

```diff
@@ -39,13 +40,30 @@ PresenceManager.prototype.setup = function() {
   });
 
   // save so you removeListener on destroy
-  this.sentryListener = function(sentry) {
+  this.sentryListener = function (sentry) {
+    if (manager.destroying) return;
```

Comment:
This is unnecessary since we do
``` this.sentry.removeListener('down', this.sentryListener);```
inside destroy.

You know what, that could act as the flag!